nwb_conversion_tools/datainterfaces/ecephys/basesortingextractorinterface.py

```diff
@@ -66,56 +67,49 @@ def run_conversion(
-            recording = se.NumpyRecordingExtractor(
-                timeseries=np.array(range(n_channels)),
-                sampling_frequency=self.sorting_extractor.get_sampling_frequency(),
-            )
-            add_devices(recording=recording, nwbfile=nwbfile, metadata=metadata)
-            add_electrode_groups(recording=recording, nwbfile=nwbfile, metadata=metadata)
-            add_electrodes(recording=recording, nwbfile=nwbfile, metadata=metadata)
+            temp_ephys = TempEcephysInterface(timeseries=np.array(range(n_channels)), sampling_frequency=1)
+            temp_ephys.run_conversion(nwbfile=nwbfile, metadata=metadata, write_electrical_series=False)
 
-        if stub_test:
-            max_min_spike_time = max(
-                [
-                    min(x)
-                    for y in self.sorting_extractor.get_unit_ids()
-                    for x in [self.sorting_extractor.get_unit_spike_train(y)]
-                    if any(x)
-                ]
-            )
-            stub_sorting_extractor = se.SubSortingExtractor(
-                self.sorting_extractor,
-                unit_ids=self.sorting_extractor.get_unit_ids(),
-                start_frame=0,
-                end_frame=1.1 * max_min_spike_time,
-            )
-            # TODO: copy over unit properties (SubRecording and SubSorting do not carry these automatically)
-            sorting_extractor = stub_sorting_extractor
-        else:
-            sorting_extractor = self.sorting_extractor
+        conversion_opts = default_export_ops()
+        conversion_opts.update(**kwargs)
+        conversion_opt_schema = default_export_ops_schema()
+        validate(instance=conversion_opts, schema=conversion_opt_schema)
+
+        self.writer_class.add_to_nwb(nwbfile, metadata, **conversion_opts)
 
-        property_descriptions = dict()
-        for metadata_column in metadata.get("Ecephys", dict()).get("UnitProperties", []):
-            property_descriptions.update({metadata_column["name"]: metadata_column["description"]})
-            for unit_id in sorting_extractor.get_unit_ids():
-                # Special condition for wrapping electrode group pointers to actual object ids rather than string names
-                if metadata_column["name"] == "electrode_group":
-                    if nwbfile.electrode_groups:
-                        sorting_extractor.set_unit_property(
-                            unit_id=unit_id,
-                            property_name=metadata_column["name"],
-                            value=nwbfile.electrode_groups[
-                                self.sorting_extractor.get_unit_property(
-                                    unit_id=unit_id, property_name="electrode_group"
-                                )
-                            ],
-                        )
-        write_sorting(sorting_extractor, property_descriptions=property_descriptions, nwbfile=nwbfile)
```

Comment:
I think this is unnecessary, since setting of any properties (`sorting_extractor.set_unit_property()`)should happen at the `__init__` . Once this is done, the `add_units()` method will automatically pull these.